### PR TITLE
upgrade ember-cli to 2.12.3

### PR DESCRIPTION
**Pull request by helloWorldCindy**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

Upgrade Ember-cli to 2.12.3

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -13,4 +13,4 @@
       }
     ]
   }
-}
+};
```
Comment (juwara0) [inline .eslintrc.js]: The semi-colon needs to be removed.

File: ember-cli-build.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* global require, module */
+/* eslint-env node */
 var EmberAddon = require('ember-cli/lib/broccoli/ember-addon')
```
Comment (juwara0) [inline ember-cli-build.js]: This should be updated to: `const EmberAddon = require('ember-cli/lib/broccoli/ember-addon')`

File: package.json [hunk 3]
```diff
@@ -27,22 +27,24 @@
   "author": "Matthew Dahl (https://github.com/sandersky)",
   "license": "MIT",
   "devDependencies": {
-    "broccoli-asset-rev": "^2.5.0",
-    "ember-cli": "^2.11.0",
+    "broccoli-asset-rev": "^2.4.5",
+    "ember-cli": "2.12.3",
     "ember-cli-app-version": "^2.0.1",
```
Comment (juwara0) [inline package.json]: This should be removed.

File: package.json [hunk 4]
```diff
@@ -27,22 +27,24 @@
...
     "ember-cli-chai": "^0.3.2",
     "ember-cli-dependency-checker": "^1.3.0",
-    "ember-cli-htmlbars-inline-precompile": "0.3.6",
+    "ember-cli-htmlbars-inline-precompile": "^0.3.6",
     "ember-cli-inject-live-reload": "^1.4.1",
     "ember-cli-mocha": "0.13.2",
+    "ember-cli-shims": "^1.0.2",
     "ember-cli-test-loader": "^1.1.1",
```
Comment (juwara0) [inline package.json:38]: This should be removed.

File: package.json [hunk 5]
```diff
@@ -27,22 +27,24 @@
...
     "ember-cli-uglify": "^1.2.0",
     "ember-disable-prototype-extensions": "^1.1.0",
-    "ember-export-application-global": "^1.1.1",
-    "ember-load-initializers": "0.6.3",
+    "ember-export-application-global": "^1.0.5",
+    "ember-load-initializers": "^0.6.0",
+    "ember-source": "~2.12.0",
     "ember-resolver": "^2.1.1",
```
Comment (juwara0) [inline package.json]: This should be `"ember-resolver": "^2.0.3",`

File: tests/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    embertest: true
+  }
+};
```
Comment (juwara0) [inline tests/.eslintrc.js]: The semi-colon needs to be removed.

Comment (helloWorldCindy) [reply tests/.eslintrc.js]: Fixed.

File: package.json (hunk 5 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: package.json (hunk 4 above)
Comment (helloWorldCindy) [reply package.json:38]: Fixed.

File: package.json (hunk 3 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

File: ember-cli-build.js (hunk 2 above)
Comment (helloWorldCindy) [reply ember-cli-build.js]: Fixed.

File: .eslintrc.js (hunk 1 above)
Comment (helloWorldCindy) [reply .eslintrc.js]: Fixed.

File: .npmignore [hunk 7]
```diff
@@ -12,6 +12,7 @@
 .github
 .gitignore
 .jshintrc
```
Comment (notmessenger) [inline .npmignore]: Remove

File: .npmignore [hunk 8]
```diff
@@ -12,6 +12,7 @@
...
+.eslintrc.js
```
Comment (notmessenger) [inline .npmignore]: Place this entry in the file alphabetically and remove the entry for `.eslintrc`

File: .travis.yml [hunk 9]
```diff
@@ -37,7 +33,8 @@ matrix:
   allow_failures:
   - env: EMBER_TRY_SCENARIO=ember-beta
   - env: EMBER_TRY_SCENARIO=ember-canary
-
+  - node_js: 'stable'
```
Comment (notmessenger) [inline .travis.yml]: Do we want this @juwara0?  I have not been adding this to my PRs

File: tests/dummy/config/environment.js [hunk 10]
```diff
@@ -1,9 +1,14 @@
+/* eslint-env node */
```
Comment (notmessenger) [inline tests/dummy/config/environment.js:1]: Also remove `ENV.rootURL = '/'` entry from test configuration

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: .npmignore (hunk 7 above)
Comment (juwara0) [reply .npmignore]: Fixed.

File: .npmignore (hunk 8 above)
Comment (juwara0) [reply .npmignore]: Fixed

File: .travis.yml (hunk 9 above)
Comment (juwara0) [reply .travis.yml]: Yes, lets keep this in for now.

File: tests/dummy/config/environment.js (hunk 10 above)
Comment (juwara0) [reply tests/dummy/config/environment.js:1]: Fixed

File: .travis.yml [hunk 11]
```diff
@@ -22,34 +22,43 @@ cache:
...
   - node_js: 'stable'
-  
+
 before_install:
-- "export DISPLAY=:99.0"
-- "sh -e /etc/init.d/xvfb start"
-- npm config set spin false
-- npm install -g coveralls pr-bumper
+  - npm config set spin false
+  - npm install -g coveralls pr-bumper@^1.0.0
```
Comment (notmessenger) [inline .travis.yml:45]: Do we want `1.0.0` or `2.0.0`?

**Review by notmessenger — APPROVED** [review]

File: .travis.yml (hunk 11 above)
Comment (juwara0) [reply .travis.yml:45]: All the rest of the repos are running `1.0.0` this one was not. I would like to make the switch to `2.0.0` a separate effort.

**Comment by juwara0** [comment]
Approved.

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/ember-validator-shim/pull-request/50/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/ember-validator-shim)

**Pull request merged**